### PR TITLE
Added samplers_params

### PR DESCRIPTION
**Pull request by TezRomacH**
## Description

- Added config key for `samplers_params`

## Related Issue

N/A

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have checked the code-style using `make check-style`.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by TezRomacH** [comment]
@Scitator @bagxi could you please add your review?

**Review by gazay — APPROVED** [review]
I don't see any problems with this code but without tests it's hard to be sure. Btw it would be great to add these sampler params to wizard as well, but I can do it in separate PR

**Review by lightforever — APPROVED** [review]
Everythings seems reasonable

File: catalyst/dl/experiment/config.py [hunk 1]
```diff
@@ -345,6 +351,17 @@ def get_loaders(self, stage: str) -> "OrderedDict[str, DataLoader]":
             assert isinstance(overridden_loader_params, dict), \
                 f"{overridden_loader_params} should be Dict"
 
+            sampler_params = samplers_params.pop(name, None)
```
Comment (Scitator) [inline catalyst/dl/experiment/config.py:354]: we should also check, that user's datasets and samplers have same names,

for example, what would be if we have
datasets: `{"train": ..., "valid": ...}` and samplers `{"train": ..., "valid000": ...}`?

Comment (Scitator) [reply catalyst/dl/experiment/config.py:354]: btw, the same logic should be implemented with loaders

Comment (TezRomacH) [reply catalyst/dl/experiment/config.py:354]: No, you don't. It's already done by default. 

You can see above that we are only iterated by the names of the datasets `for name, ds_ in datasets.items()`. And for each of them, we're trying to get a loader and a sampler

Comment (Scitator) [reply catalyst/dl/experiment/config.py:354]: please, add a validation check for our Catalyst.todo for this case

**Pull request merged**